service/service.py

```diff
@@ -289,11 +279,14 @@ def get(self):
             "active",
             "product",
         ]
-        app.logger.info(request.args)
-        if any(i in request.args for i in filters):
-            promotions = Promotion.find_by_query_string(request.args)
+
+        args = promotion_args.parse_args()
+        app.logger.info(args)
+        if any(i in args for i in filters):
+            promotions = Promotion.find_by_query_string(args)
         else:
```

Comment:
I think this else might never get reached? which might be why the test coverage is down

Reply:
Thanks for pointing this out. It's now fixed. 

With the `RequestParser()`, we'll always have a dictionary with all the possible args. If the arg wasn't in the query string, the value would just be `None`. Here's what we'll get when GET `/promotions` with empty query string:

`{'title': None, 'promo_code': None, 'promo_type': None, 'amount': None, 'start_date': None, 'end_date': None, 'duration': None, 'active': None, 'is_site_wide': None, 'product': None} `